third_party/docfx-doclet-143274/src/main/java/com/microsoft/build/YmlFilesBuilder.java

```diff
@@ -253,6 +263,106 @@ MetadataFileItem buildMetadataFileItem(Element element) {
+            String methodName = "#" + nameSplit.get(nameSplit.size() - 1);
+
+            String argumentsName = argumentSplit.get(1).replaceAll("[,)]", "-");
+
+            // endURL: "java/lang/Object.html#equals-java.lang.Object-"
+            endURL = className + ".html" + methodName + "-" + argumentsName;
+        } else {
+            // example2
+            // endURL = java/lang/Object.html
+            endURL = endURL.replaceAll("\\.", "/");
+            endURL = endURL + ".html";
+        }
+        return JAVA_BASE_URL + endURL;
+    }
+
+    private void updateExternalReferences(List<MetadataFile> classMetadataFiles) {
+        classMetadataFiles.forEach(file -> file.getReferences()
+                .forEach(ref -> updateExternalReference(ref)));
+    }
+
+    private void updateExternalReference(MetadataFileItem reference) {
+        String uid = reference.getUid();
+        uid = updateReferenceUid(uid);
+
+        if (isJavaPrimitive(uid)) {
+            reference.setHref(PRIMITIVE_URL);
+            return;
+        }
+        if (isJavaLibrary(uid)) {
+            reference.setHref(getJavaReferenceHref(uid));
+        }
+        if (isExternalReference(uid)) {
+            reference.setIsExternal(true);
+        }
+        if (reference.getSpecForJava().size() > 0) {
+            for (SpecViewModel spec : reference.getSpecForJava()) {
+                String specUid = spec.getUid();
+                if (specUid != null) {
+                    if (isJavaPrimitive(specUid)) {
+                        spec.setHref(PRIMITIVE_URL);
+                    }
+                    if (isJavaLibrary(specUid)) {
+                        spec.setHref(getJavaReferenceHref(specUid));
+                    }
+                    if (isExternalReference(specUid)) {
+                        spec.setIsExternal(true);
+                    }
+                }
+            }
+        }
+    }
+
+    private String updateReferenceUid(String uid){
+        if (ENDING_PATTERN.matcher(uid).find()) {
+            uid = uid.replace("<?>","");
+        }
+        return uid;
+    }
+
+    private boolean isExternalReference(String uid) {
+        if (PROTOBUF_PATTERN.matcher(uid).find() || GAX_PATTERN.matcher(uid).find() || APICOMMON_PATTERN.matcher(uid).find() || GAX_PATTERN.matcher(uid).find() || LONGRUNNING_PATTERN.matcher(uid).find()) {
+            return true;
+        }
+        return false;
+    }
+
+     private boolean isJavaPrimitive(String uid) {
+        if (uid.equals("boolean") || uid.equals("int") || uid.equals("byte") || uid.equals("long") || uid.equals("float") || uid.equals("double") || uid.equals("char") || uid.equals("short")) {
+            return true;
+        }
+        return false;
+    }
+
+    private boolean isJavaLibrary(String uid) {
+        if (JAVA_PATTERN.matcher(uid).find()) {
+            return true;
+        }
+        return false;

```

Comment:
Consider:
```suggestion
        return JAVA_PATTERN.matcher(uid).find()
```

Same pattern applies above.